### ISSUE TITLE
validate returns an error

### ISSUE THREAD

**Issue by gurmukhp** [voltrue2/in-app-purchase#8]
I am trying to use your module and have copied in the sample code you've given:

```
	iap.setup(function (error) {
	    if (error) {
	        // error hmm 
	    }
	    iap.validate(iap.APPLE, function (error, response) {
	        if (error) {
	            // error 
	        }
	        if (iap.isValidated(response)) {
	            var purcahseDataList = iap.getPurchaseData(response);
	            /*
	                [
	                    {
	                        productId: xxx,
	                        purchasedDate: yyy,
	                        quantity: zzz
	                    }
	                ]
	            */
	        }
	    });
	});
```

However I'm getting the error about validate?

```
[ERROR] TypeError: undefined is not a function
    at /Users/gurmukh.panesar/Sites/iap/node_modules/in-app-purchase/index.js:45:11
    at handleResponse (/Users/gurmukh.panesar/Sites/iap/node_modules/in-app-purchase/lib/apple.js:115:2)
    at /Users/gurmukh.panesar/Sites/iap/node_modules/in-app-purchase/lib/apple.js:78:3
    at Request._callback (/Users/gurmukh.panesar/Sites/iap/node_modules/in-app-purchase/lib/apple.js:129:10)
    at Request.self.callback (/Users/gurmukh.panesar/Sites/iap/node_modules/in-app-purchase/node_modules/request/request.js:122:22)
    at Request.EventEmitter.emit (events.js:98:17)
    at Request.<anonymous> (/Users/gurmukh.panesar/Sites/iap/node_modules/in-app-purchase/node_modules/request/request.js:888:14)
    at Request.EventEmitter.emit (events.js:117:20)
    at IncomingMessage.<anonymous> (/Users/gurmukh.panesar/Sites/iap/node_modules/in-app-purchase/node_modules/request/request.js:839:12)
    at IncomingMessage.EventEmitter.emit (events.js:117:20)
```

I'm using Express 3.0, if that's any help

**Comment by voltrue2** [comment]
hi thanks for the report. what version of in-app-purchase are you using?

**Comment by voltrue2** [comment]
by the way it looks like your iap.validate() function is missing an argument.

it needs the following:

iap.validate(iap.APPLE, receipt, callbackFunction);

I think that is what the problem is. Could you have a look at your code?

**Comment by gurmukhp** [comment]
Hey @voltrue2 
thanks so much for looking into this, it looked like I was missing a param to the validate function, just added it and it works now :+1: 

Thank you again

**Comment by voltrue2** [comment]
Awesome. Glad it's working for you now. Thanks for reporting this. The documentation has been corrected!

On Wed, Apr 22, 2015 at 10:11 PM, voltrue2/in-app-purchase
<reply@reply.github.com>
wrote:
Closed #8 [https://github.com/voltrue2/in-app-purchase/issues/8] .

—
Reply to this email directly or view it on GitHub
[https://github.com/voltrue2/in-app-purchase/issues/8#event-287228162] .[https://github.com/notifications/beacon/ACKY4R-Hr5F30ExZKkd0UPg5T1BP7MS4ks5oB5WRgaJpZM4EF-e9.gif]
